### PR TITLE
Update samples and tests to target netcoreapp2.1

### PR DESCRIPTION
**Pull request by pranavkm**

### CODE REVIEW

File: test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj [hunk 1]
```diff
@@ -1,15 +1,14 @@
...
-    <TargetFrameworks>netcoreapp2.0;net461</TargetFrameworks>
-    <TargetFrameworks Condition="'$(OS)' != 'Windows_NT'">netcoreapp2.0</TargetFrameworks>
+    <TargetFrameworks>$(StandardTestTfms)</TargetFrameworks>
 
     <!--
       Workaround for "Use executable flags in Microsoft.NET.Test.Sdk" (https://github.com/Microsoft/vstest/issues/792).
       Remove when fixed.
     -->
     <HasRuntimeOutput>true</HasRuntimeOutput>
-    <RuntimeIdentifier Condition="'$(TargetFramework)' != 'netcoreapp2.0'">win7-x64</RuntimeIdentifier>
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: Running this for x64 produces an error that says it mismatches the test platform architecture.

Comment (moozzyk) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: Is it a regression then? Because it used to work just fine...

Comment (BrennanConroy) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: What

File: test/Directory.Build.props [hunk 2]
```diff
@@ -1,6 +1,13 @@
 ﻿<Project>
   <Import Project="..\Directory.Build.props" />
 
+  <PropertyGroup>
+    <DeveloperBuildTestTfms>netcoreapp2.1</DeveloperBuildTestTfms>
+    <StandardTestTfms>$(DeveloperBuildTestTfms)</StandardTestTfms>
+    <StandardTestTfms Condition=" '$(DeveloperBuild)' != 'true' ">netcoreapp2.1;netcoreapp2.0</StandardTestTfms>
```
Comment (BrennanConroy) [inline test/Directory.Build.props]: Why is netcoreapp2.0 here?

Comment (davidfowl) [reply test/Directory.Build.props]: We run tests against all

File: test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj (hunk 1 above)
Comment (pranavkm) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: https://github.com/aspnet/BuildTools/issues/471

Comment (pranavkm) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: If it's super important to target the tests to run on x64, I can create a follow up work item to revert this once https://github.com/aspnet/BuildTools/issues/471 is resolved.

Comment (muratg) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: @pranavkm yes lets file a bug to bring them back please.

Comment (pranavkm) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: https://github.com/aspnet/SignalR/issues/1133

Comment (moozzyk) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: I am not concerned about the tests too much as long as they run on full framework. The thing I am concerned about (not understanding the root cause) is whether customers will have the same issue with their projects targeting full framework.

Comment (pranavkm) [reply test/Microsoft.AspNetCore.SignalR.Tests/Microsoft.AspNetCore.SignalR.Tests.csproj]: It's a change in how we run the tests in our SDK. It's not an issue with the actual binary.

**Pull request merged**